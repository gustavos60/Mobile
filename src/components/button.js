import React from 'react';
import PropTypes from 'prop-types';

import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  ActivityIndicator,
  ViewPropTypes,
} from 'react-native';
import colors from '../colors';

const Button = ({
  label,
  testID,
  onPress,
  loading,
  disabled,
  labelStyle,
  buttonStyle,
}) => (
  <TouchableOpacity
    testID={testID}
    onPress={onPress}
    style={[styles.button, buttonStyle]}
    activeOpacity={0.7}
    disabled={disabled}
  >
    <View style={styles.container}>
      <Text style={[styles.label, labelStyle]}>
        {label}
      </Text>
      {
        loading ? (
          <ActivityIndicator
            size="small"
            color="white"
            style={styles.loading}
          />
        ) : null
      }
    </View>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    width: 150,
    height: 55,
    borderRadius: 30,
    backgroundColor: colors.primary,
  },
  label: {
    fontSize: 16,
    color: colors.white,
  },
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  loading: {
    position: 'absolute',
    right: 15,
    bottom: 0,
    top: 0,
  },
});

Button.propTypes = {
  label: PropTypes.string,
  loading: PropTypes.bool,
  testID: PropTypes.string,
  disabled: PropTypes.bool,
  labelStyle: ViewPropTypes.style,
  buttonStyle: ViewPropTypes.style,
  onPress: PropTypes.func,
};

Button.defaultProps = {
  label: '',
  testID: '',
  loading: false,
  disabled: false,
  labelStyle: null,
  buttonStyle: null,
  onPress: () => {},
};

export default Button;
