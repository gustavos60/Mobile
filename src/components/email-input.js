import React from 'react';
import PropTypes from 'prop-types';

import {
  View,
  TextInput,
  StyleSheet,
  Dimensions,
} from 'react-native';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '../colors';

const DEVICE_WIDTH = Dimensions.get('screen').width;

const EmailInput = ({
  valid,
  email,
  inputTestId,
  onChangeText,
}) => (
  <View
    style={[
      styles.container,
      { borderColor: valid ? colors.gray : colors.secondary },
    ]}
  >
    <Icon
      style={styles.icon}
      name="email"
      size={30}
      color="gray"
    />
    <TextInput
      testID={inputTestId}
      value={email}
      keyboardType="email-address"
      onChangeText={onChangeText}
      placeholder="Email"
      style={styles.input}
      autoCompleteType="email"
    />
  </View>
);

const styles = StyleSheet.create({
  input: {
    padding: 10,
    fontSize: 18,
    flex: 1,
  },
  container: {
    margin: 10,
    width: DEVICE_WIDTH * 0.80,
    borderWidth: 1,
    borderColor: colors.gray,
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
  },
  icon: {
    marginLeft: 10,
  },
});

EmailInput.propTypes = {
  valid: PropTypes.bool,
  inputTestId: PropTypes.string,
  email: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
};

EmailInput.defaultProps = {
  valid: true,
  inputTestId: '',
};

export default EmailInput;
