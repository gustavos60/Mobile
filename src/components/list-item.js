import React from 'react';
import PropTypes from 'prop-types';

import {
  View,
  Text,
  Image,
  StyleSheet,
} from 'react-native';

const ListItem = ({
  place,
}) => (
  <View style={styles.container}>
    <Image style={styles.image} source={{ uri: place.uri }} />
    <View>
      <Text style={styles.name}>
        {place.name}
      </Text>
      <Text style={styles.text}>
        Latitude:
        {' '}
        {place.latitude}
      </Text>
      <Text style={styles.text}>
        Longitude:
        {' '}
        {place.longitude}
      </Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: 100,
    flexDirection: 'row',
    marginBottom: 10,
  },
  image: {
    height: '90%',
    width: '30%',
    borderRadius: 10,
    marginRight: 10,
  },
  text: {
    marginBottom: 5,
  },
  name: {
    fontSize: 18,
    marginBottom: 5,
    overflow: 'visible',
  },
});

ListItem.propTypes = {
  place: PropTypes.shape({
    uri: PropTypes.string,
    name: PropTypes.string,
    latitude: PropTypes.number,
    longitude: PropTypes.number,
  }),
};

ListItem.defaultProps = {
  place: null,
};

export default ListItem;
