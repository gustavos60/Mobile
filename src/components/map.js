import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MapView, { Marker } from 'react-native-maps';

export const GoogleMap = ({
  region,
  places,
}) => (
  <MapView
    initialRegion={region}
    style={{ flex: 1 }}
    maxZoomLevel={40}
  >
    {places.map((place) => (
      <Marker
        key={`${place.name}+${place.latitude}`}
        title={place.name}
        coordinate={{
          latitude: place.latitude,
          longitude: place.longitude,
        }}
      />
    ))}
  </MapView>
);

GoogleMap.propTypes = {
  region: PropTypes.shape({
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    latitudeDelta: PropTypes.number,
    longitudeDelta: PropTypes.number,
  }).isRequired,
  places: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]),
};

GoogleMap.defaultProps = {
  places: null,
};

const mapStateToProps = (state) => ({
  places: state.places.places,
});

export default connect(mapStateToProps)(GoogleMap);
