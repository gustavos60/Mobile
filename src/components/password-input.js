import React, { useState } from 'react';
import PropTypes from 'prop-types';

import {
  View,
  TextInput,
  StyleSheet,
  Dimensions,
} from 'react-native';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '../colors';

const DEVICE_WIDTH = Dimensions.get('screen').width;

const PasswordInput = ({
  valid,
  password,
  inputTestId,
  onChangeText,
}) => {
  const [visible, setVisible] = useState(false);
  return (
    <View
      style={[
        styles.container,
        { borderColor: valid ? colors.gray : colors.secondary },
      ]}
    >
      <Icon
        style={styles.lockIcon}
        name="lock"
        size={30}
        color="gray"
      />
      <TextInput
        testID={inputTestId}
        value={password}
        secureTextEntry={!visible}
        onChangeText={onChangeText}
        placeholder="Senha"
        style={styles.input}
        autoCompleteType="password"
      />
      <Icon
        testID="password-toggle-visibility"
        style={styles.eyeIcon}
        name="eye"
        size={30}
        onPress={() => setVisible((prevVisible) => !prevVisible)}
        color={visible ? colors.black : colors.gray}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  input: {
    padding: 10,
    fontSize: 18,
    flex: 1,
  },
  container: {
    margin: 10,
    width: DEVICE_WIDTH * 0.80,
    borderWidth: 1,
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
  },
  lockIcon: {
    marginLeft: 10,
  },
  eyeIcon: {
    marginRight: 10,
  },
});

PasswordInput.propTypes = {
  valid: PropTypes.bool,
  inputTestId: PropTypes.string,
  password: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
};

PasswordInput.defaultProps = {
  valid: true,
  inputTestId: '',
};

export default PasswordInput;
