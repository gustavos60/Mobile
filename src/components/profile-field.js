import React from 'react';
import PropTypes from 'prop-types';

import {
  View,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';

import colors from '../colors';

const ProfileField = ({
  label,
  value,
  valid,
  testID,
  canEdit,
  placeholder,
  onChangeText,
  textInputProps,
}) => (
  <View style={styles.container}>
    <Text style={styles.label}>
      {label}
    </Text>
    {
      canEdit ? (
        <View
          style={[
            styles.inputContainer,
            { borderColor: valid ? colors.gray : colors.secondary },
          ]}
        >
          <TextInput
            value={value}
            testID={testID}
            style={styles.input}
            placeholder={placeholder}
            onChangeText={onChangeText}
            {...textInputProps}
          />
        </View>
      ) : <Text style={styles.input}>{value === '' ? '- -' : value}</Text>
    }

  </View>
);

const styles = StyleSheet.create({
  label: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 5,
  },
  input: {
    fontSize: 18,
  },
  inputContainer: {
    borderWidth: 1,
    borderColor: colors.gray,
    borderRadius: 10,
  },
  container: {
    padding: 10,
  },
});

ProfileField.propTypes = {
  valid: PropTypes.bool,
  canEdit: PropTypes.bool,
  onChangeText: PropTypes.func,
  testID: PropTypes.string,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  textInputProps: PropTypes.shape(),
};

ProfileField.defaultProps = {
  testID: '',
  valid: true,
  canEdit: false,
  onChangeText: () => {},
  textInputProps: null,
};

export default ProfileField;
