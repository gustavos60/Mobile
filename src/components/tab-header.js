import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { StackActions, NavigationActions } from 'react-navigation';

import { logout } from '../redux/profile/actions';

import colors from '../colors';

export const TabHeader = ({
  title,
  testID,
  navigation,
  logoutAction,
}) => (
  <View style={styles.header}>
    <Text
      style={styles.title}
      testID={testID}
    >
      {title}
    </Text>
    <TouchableOpacity
      testID="logoutButton"
      style={styles.logout}
      activeOpacity={0.7}
      onPress={() => {
        logoutAction();
        navigation.dispatch(StackActions.reset({
          index: 0,
          actions: [NavigationActions.navigate({ routeName: 'Auth' })],
        }));
      }}
    >
      <Icon
        name="logout"
        color={colors.white}
        size={30}
      />
    </TouchableOpacity>
  </View>
);

TabHeader.propTypes = {
  title: PropTypes.string,
  testID: PropTypes.string,
  logoutAction: PropTypes.func.isRequired,
};

TabHeader.defaultProps = {
  title: '',
  testID: '',
};

const styles = StyleSheet.create({
  header: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.primary,
    height: 90,
  },
  title: {
    fontSize: 28,
    color: colors.white,
    fontWeight: 'bold',
  },
  logout: {
    position: 'absolute',
    right: 20,
    top: 30,
  },
});

const mapDispatchToProps = (dispatch) => ({
  logoutAction: () => dispatch(logout()),
});

export default connect(null, mapDispatchToProps)(TabHeader);
