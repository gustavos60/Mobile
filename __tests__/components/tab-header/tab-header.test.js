import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { TabHeader } from '../../../src/components/tab-header';

const mockedProps = {
  title: 'Perfil',
  testID: 'perfilHeader',
  navigation: {
    dispatch: jest.fn(),
  },
  logoutAction: jest.fn(),
};

describe('TabHeader tests', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<TabHeader {...mockedProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
