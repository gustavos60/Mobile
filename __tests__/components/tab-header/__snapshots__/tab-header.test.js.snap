// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabHeader tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#123456",
      "height": 90,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
    testID="perfilHeader"
  >
    Perfil
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "top": 30,
      }
    }
    testID="logoutButton"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
